evm/src/test/java/com/horizen/evm/StateDBTest.java

```diff
@@ -0,0 +1,167 @@
+package com.horizen.evm;
+
+import com.horizen.evm.utils.Converter;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+
+import static org.junit.Assert.*;
+
+public class StateDBTest extends LibEvmTestBase {
```

Comment:
I think the Unit tests coverage is not enough.
We miss tests for `proofs`, `logs`, etc.
Ensure that every method of StateDB has been tests carefully.